### PR TITLE
Compatible for FileAndType.

### PR DESCRIPTION
**Pull request by vwxyzh**
@herohua 
#449

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs [hunk 1]
```diff
@@ -847,6 +847,10 @@ private IEnumerable<ManifestItemWithContext> ExportManifest(HostService hostServ
                         {
                             Logger.LogDiagnostic($"Processor {hostService.Processor.Name}: Saving...");
                             m.BaseDir = context.BuildOutputFolder;
+                            if (m.PathRewriter != null)
+                            {
+                                m.File = m.PathRewriter(m.File);
+                            }
                             if (m.FileAndType.SourceDir != m.FileAndType.DestinationDir)
```
Comment (herohua) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: move above `if` below `if (m.FileAndType.SourceDir != m.FileAndType.DestinationDir)`?

**Comment by herohua** [comment]
:+1:

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: only one can be occurred.

---
In reply to: [78318231](https://github.com/dotnet/docfx/pull/688#discussion_r78318231) [](ancestors = 78318231)

**Comment by hellosnow** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**